### PR TITLE
[doc-apptools] Add developer guide and user guide for apptools

### PR DESCRIPTION
**Pull request by yunxliu**
- Add developer guide for apptools testsuit
- Add user guide for apptools testsuit

### CODE REVIEW

File: doc/Apptools_Test_Suite_Developer_Guide.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Apptools Test Suite Developer Guide
+
+## Overview
+
+This document is intended for developers who contribute Apptools test cases development.
```
Comment (Honry) [inline doc/Apptools_Test_Suite_Developer_Guide.md]: This document is intended for those who want to make test contributions to App-Tools test suite.

File: doc/Apptools_Test_Suite_Developer_Guide.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+Please formiliarize yourself with the following knowledge:
+
+- Basic knowledge in [web test suite developer guide](./Web_Test_Suite_Developer_Guide.md)
+- [PyUnit](http://pyunit.sourceforge.net/pyunit.html), the standard unit testing framework for Python. It is called by testkit-lite to support the test execution.
+
+## Test Case File Template
+
+Auto tests is using python language, and auto testCase file template looks like this:
```
Comment (Honry) [inline doc/Apptools_Test_Suite_Developer_Guide.md]: 's/is/are'

File: doc/Apptools_Test_Suite_Developer_Guide.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    def subcase3(self):
+
+if __name__ == '__main__':
+    unittest.main()
+```
+
+- "def" represents one subcase, and it requires a blank line between each subcases
+- Similar test points can be put in one testCase file
+- One testCase file may includes one or multiple subcases
```
Comment (Honry) [inline doc/Apptools_Test_Suite_Developer_Guide.md]: 's/testCase/test case'.

File: doc/Apptools_Test_Suite_Developer_Guide.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Apptools Test Suite Developer Guide
```
Comment (Honry) [inline doc/Apptools_Test_Suite_Developer_Guide.md]: 's/Apptools/App-tools/', please also update others.

File: doc/Apptools_Test_Suite_Developer_Guide.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+## How to contribute new cases for Apptools?
+
+1. Design new test case according to the specification and/or feature requirement, and add new case information to `tests.full.xml`. The `tests.xml` can be converted from `tests.full.xml` by [xmlsimplifier](https://github.com/crosswalk-project/crosswalk-test-suite/tree/master/tools/xmlsimplifier).
+1. Develop test script following the [test case naming convention](./Web_Test_Suite_Developer_Guide.md#Test-case-naming-convention) and [test case coding style](./Web_Test_Suite_Developer_Guide.md#test-case-coding-style), and put it under `<test-case-sub-directory>`.
+
+Note:
+
+- Each test should have an entry HTML file/Python script.
+- The test script can be embedded into the HTML/Python file or be used as separated JavaScript/Python file.
+- Set `<set type='pyunit'>` in `tests*.xml` for PyUnit test.
```
Comment (Honry) [inline doc/Apptools_Test_Suite_Developer_Guide.md]: You can add one more section to introduce App-tools in-short.

File: doc/Apptools_Test_Suite_User_Guide.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
+# Apptools Test Suite User Guide
+
+## Introduction
+
+This document provides method to run Apptools TestSuite. Currently the target platforms are Android, iOS, Linux and Windows
```
Comment (Honry) [inline doc/Apptools_Test_Suite_User_Guide.md]: Please add period at the end of each sentence.

File: doc/Apptools_Test_Suite_User_Guide.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+## Setup Environment
+
+### Linux host
```
Comment (Honry) [inline doc/Apptools_Test_Suite_User_Guide.md]: The first letter of each word in title should be capital, except conj such as and, of ...

File: doc/Apptools_Test_Suite_User_Guide.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+#### Android Platform
+
+1. Install Python,version is 2.7.6
+1. Install Android SDK ,JDK, git
+1. Install apache-ant-1.9.6 (Download from http://www.onlinedown.net/softdown/77637_2.htm)
+1. Install node and npm (node version is 0.10.30,npm version is 1.4.21)
+1. Install setuptools module (Download from https://pypi.python.org/pypi/setuptools)
+1. Modify the ~/.bashrc
```
Comment (Honry) [inline doc/Apptools_Test_Suite_User_Guide.md]: Only need to tell user to setup the environment variable for each path.

File: doc/Apptools_Test_Suite_User_Guide.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+$ cd [testprefix-path]/opt/apptools-linux-tests/
+$ testkit-lite -f $PWD/tests.xml -A --comm deepin --testprefix $PWD/../../ -o $PWD/result.xml
+```
+### Windows Platform
+```xml
+$ cd [testprefix-path]/opt/apptools-windows-tests/
+$ testkit-lite -f [testprefix-path]/opt/apptools-windows-tests/tests.xml -A -o [testprefix-path]/opt/apptools-windows-tests/result.xml --comm localhost --testprefix=[testprefix-path] --non-active
+```
+
+## Detail introduction and precondition for AppTools
```
Comment (Honry) [inline doc/Apptools_Test_Suite_User_Guide.md]: This section is unnecessary.

**Comment by ibelem** [comment]
LGTM

**Pull request merged**